pkg/v3/resource/namespace/update.go

```diff
@@ -3,13 +3,36 @@ package namespace
 import (
 	"context"
 
+	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/controller"
 )
 
+// ApplyUpdateChange is a no-op because the namespace in the guest cluster is
+// not updated.
 func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange interface{}) error {
 	return nil
 }
 
 func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*controller.Patch, error) {
+	create, err := r.newCreateChange(ctx, obj, currentState, desiredState)
```

Comment:
Just out of interest: Why do we need to implement `NewUpdatePatch` if we don't update the namespace?

Reply:
We can't reliably distinguish between create and update events from the informer. So operatorkit treats both the same. This is why there is NewUpdatePatch but no NewCreatePatch.

Reply:
This is to make sure things are idempotent. You can think of it as `CreateOrUpdate`.